### PR TITLE
Update web API node to 3.1 or later

### LINKED ISSUE #24695: Drop 2.x content and sample apps
Rick, I have a few minutes this AM ... I'll try a quick-ish pass on the node to see if I can fix up the 2.x aspects.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b391017f-25e1-d5e1-c4f7-020408ade7b5
* Version Independent ID: 263a7b64-2e42-bbfc-c1d3-61680438e1e4
* Content: [Create web APIs with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-3.1)
* Content Source: [aspnetcore/web-api/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/web-api/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-webapi**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **scaddie**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #24695

😝 *I'm surprised this built on the first pass w/o warnings!*

This is a quick-ish 🏃 pass to the web API node for text and sample updates (e.g., drop 2.x, move 3.0 to 3.1, update topic versions).

This isn't a total overhaul. Given the size of the node, I don't have time to ...

* Update API doc cross-links.
* Code-fence all of the nits and bits.
* Completely revise sample folders to the correct repo format (e.g., some of them are just in `sample`/`samples`, not `samples/3.x` in preparation for `samples/5.x`, `samples/6.x`, etc.).
* Table beautification 🌷.
* Test the updated apps. **_They compile, but I can't say if they run properly._** One of them is tossing a ...

  > warning AD0001: Analyzer 'Microsoft.AspNetCore.Mvc.Api.Analyzers.ApiConventionAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [C:\Users\guard\Documents\GitHub\AspNetCore.Docs\aspnetcore\web-api\advanced\conventions\sample\ApiConventions.csproj]

  Perhaps, that's what it's supposed to throw. We can address that one on this PR if you wish. It's for the *Use web API conventions* topic ...

  https://docs.microsoft.com/aspnet/core/web-api/advanced/conventions

### CODE REVIEW

**Comment by serpent5** [comment]
I'll have a quick look at the `ApiConventionAnalyzer` issue now.

**Comment by guardrex** [comment]
We could fix it up in a follow-up PR, too. We'd review+merge, then you'd have a fresh set of updates to work from on a new branch of `main`. Also, feel free to directly edit this PR's files if that's quicker 🏃.

**Comment by serpent5** [comment]
Ok, cool. I'll just investigate for now, to see if I can find out what's going wrong. I'll report back and we can decide where to go from there.

**Comment by guardrex** [comment]
I know Rick will be along later. I doubt he'll have anything significant on this one. It's our typical set of updates here. We can patch update anything he wants to touch-up or revert later. You could review the whole PR quickly-ish ... **_It's another EYE DROPS_** 👁️💧 **_PR!_** :smile: ... then we could go ahead and merge this right after your review ... freeing you to work directly on a branch of your own. The samps are soooooo much easier to work that way imo.

**Comment by serpent5** [comment]
I can see what the problem is, and I think it's easy enough to get it into your PR if you don't mind. I can make a couple suggestions that'll fix it, and you can make the decision on whether to pull it in here... That alright?

**Comment by guardrex** [comment]
Sure thing.

Ok ... I'll react to any feedback and get this merged. I'll switch reviewer now.

Rick ... Hit me up later for any patches/reversions.

File: aspnetcore/web-api/action-return-types.md [hunk 1]
```diff
@@ -10,61 +11,41 @@ uid: web-api/action-return-types
...
-```csharp
-public async Task<IEnumerable<Product>> GetOnSaleProducts() =>
-    await _context.Products.Where(p => p.IsOnSale).ToListAsync();
-```
-
-In ASP.NET Core 3.0 and later, returning `IAsyncEnumerable<T>` from an action:
+Returning `IAsyncEnumerable<T>` from an action:
 
 * No longer results in synchronous iteration.
 * Becomes as efficient as returning <xref:System.Collections.Generic.IEnumerable%601>.
```
Comment (serpent5) [inline aspnetcore/web-api/action-return-types.md:46]: Should we change this slightly? The "no longer" is comparing it to 2.x, I think.

File: aspnetcore/web-api/action-return-types.md [hunk 2]
```diff
@@ -10,61 +11,41 @@ uid: web-api/action-return-types
 ---
 # Controller action return types in ASP.NET Core web API
 
-By [Scott Addie](https://github.com/scottaddie)
```
Comment (serpent5) [inline aspnetcore/web-api/action-return-types.md:13]: Should I also be removing authors as I update topics?

**Review by serpent5 — APPROVED** [review]
Great stuff, as always.

File: aspnetcore/web-api/action-return-types.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/web-api/action-return-types.md:46]: Yes ... I didn't really read every line of these .. **_obvously_** 🙈 As usual, I have *a gazillion* Blazor things to get back to. 😄

I'll remove that line.

File: aspnetcore/web-api/action-return-types.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/web-api/action-return-types.md:13]: Recently, Rick implied that I didn't quite have the right rule in mind or that it had changed. IIRC the original idea was that FTEs/contractors would go away unless ...

* There are community authors on the topic who aren't responsible for most of the text. The FTE/contractor did most of the work, so the byline shouldn't indicate "By {COMMUNITY AUTHOR}" ... it's misleading.
* The FTE/contractor objected to having their name removed for some reason.

I think my removals here are ok, but I can revert later back to them if the team wants to keep them on these. I'll be 👂 for that feedback.

**Comment by guardrex** [comment]
I think that whole block can be removed.

However, I'm not sure if (or what) to say about buffering there. I'm not sure how important that original remark is. I'll leave that you/Rick to decide and/or ping Pranav to ask about.

**Comment by guardrex** [comment]
I'll hold here just a sec to hear back on that. 👂 .......... then merge this.

**Comment by guardrex** [comment]
> Great stuff, as always.

Thx ... I would quote you on that to others ... **_but I don't want to see you BANNED from the Internet ... for LIFE!_** 🤣

**Comment by serpent5** [comment]
Go ahead and merge. I'll revisit the question about the block being removed in the follow-up PR we've discused.

**Pull request merged**